Integrations/RemoteFileManagement/RemoteFileManagement.py

```diff
@@ -0,0 +1,522 @@
+    'sshkey') if 'credentials' in AUTHENTICATION and 'sshkey' in AUTHENTICATION.get('credentials') and len(
+    AUTHENTICATION.get('credentials').get('sshkey')) > 0 else None
+
+SSH_EXTRA_PARAMS = demisto.params().get('ssh_extra_params').split() if demisto.params().get('ssh_extra_params',
+                                                                                            None) else None
+SCP_EXTRA_PARAMS = demisto.params().get('scp_extra_params').split() if demisto.params().get('scp_extra_params',
+                                                                                            None) else None
+DOCUMENT_ROOT = '/' + demisto.params().get('document_root') if demisto.params().get('document_root', None) else None
+
+''' UTILS '''
+
+
+def create_certificate_file(password: str, certificate: str):
+    cert_file = tempfile.NamedTemporaryFile()
+    if certificate:
+        with open(cert_file.name, "w") as f:
+            f.write(certificate)
+        os.chmod(cert_file.name, 0o400)
+    elif password:
+        # check that password field holds a certificate and not a password
+        if password.find('-----') == -1:
+            return_error('Password parameter must contain a certificate.')
+        # split certificate by dashes
+        password_list = password.split('-----')
+        # replace spaces with newline characters
+        password_fixed = '-----'.join(password_list[:2] + [password_list[2].replace(' ', '\n')] + password_list[3:])
+        with open(cert_file.name, "w") as f:
+            f.write(password_fixed)
+        os.chmod(cert_file.name, 0o400)
+    else:
+        return_error('Provide a certificate in order to connect to the remote server.')
+
+    return cert_file
+
+
+def ssh_execute(command: str):
+    if PORT and SSH_EXTRA_PARAMS:
+        param_list = ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name, '-p',
+                      PORT] + SSH_EXTRA_PARAMS + [USERNAME + '@' + HOSTNAME, command]
+        result = subprocess.run(param_list, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    elif PORT:
+        result = subprocess.run(
+            ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name, '-p', PORT,
+             USERNAME + '@' + HOSTNAME, command], stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    elif SSH_EXTRA_PARAMS:
+        param_list = ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name] + SSH_EXTRA_PARAMS + [
+            USERNAME + '@' + HOSTNAME, command]
+        result = subprocess.run(param_list, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+    else:
+        result = subprocess.run(
+            ['ssh', '-o', 'StrictHostKeyChecking=no', '-i', CERTIFICATE_FILE.name, USERNAME + '@' + HOSTNAME, command],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+    if result.stderr and result.stderr.find("Warning: Permanently added") == -1:
```

Comment:
We should check also result.returncode. Stderr can be empty and still fail.

Reply:
@glicht please see my related fixes

Reply:
Looks good